### PR TITLE
Make generating make targets more  systematic

### PR DESCRIPTION
**Pull request by obnoxxx**
**Description of changes:** 

As suggested by @BlaineEXE in  a review comment ( https://github.com/rook/rook/pull/14700#discussion_r1815351982),

the goal  of this PR is to make all generating make targets more systematic.

Concretely, the following targets are added, as aliases of corresponding existing targets:

* `gen.codegen`
*  `gen.rbac`
* `gen.docs`
* `gen.crds`
* `gen.crd-docs`

All generating targets are run by a new target `generate`.

<!-- Uncomment this section with the issue number if an issue is being resolved

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -218,11 +223,14 @@ docs-preview: ## Preview the documentation through mkdocs
...
 
+gen.crd-docs: generate-docs-crds
 generate-docs-crds: ## Build the documentation for CRD
 	@GOBIN=$(GOBIN) build/crds/generate-crd-docs.sh
 
+generate: gen.codegen gen.crds gen.rbac gen.docs gen.crd-docs
+
 .PHONY: all build.common
-.PHONY: build build.all install test check vet fmt codegen mod.check clean distclean prune
+.PHONY: build build.all install test check vet fmt codegen gen.codegen gen.rbac gen.crds gen.crd-docs gen.docs generate    mod.check clean distclean prune
```
Comment (BlaineEXE) [inline Makefile:233]:
```suggestion
.PHONY: build build.all install test check vet fmt codegen gen.codegen gen.rbac gen.crds gen.crd-docs gen.docs generate mod.check clean distclean prune
```

File: Makefile [hunk 2]
```diff
@@ -218,11 +223,14 @@ docs-preview: ## Preview the documentation through mkdocs
 docs-build:  ## Build the documentation to the `site/` directory
 	mkdocs build --strict
 
+gen.crd-docs: generate-docs-crds
 generate-docs-crds: ## Build the documentation for CRD
 	@GOBIN=$(GOBIN) build/crds/generate-crd-docs.sh
 
+generate: gen.codegen gen.crds gen.rbac gen.docs gen.crd-docs
```
Comment (BlaineEXE) [inline Makefile:230]: Let's add some doctext to the top level command to help users as well. I'm less concerned about doctext for other `gen.` targets, and they could even make the help text a little too verbose.

```suggestion
generate: gen.codegen gen.crds gen.rbac gen.docs gen.crd-docs ## DOCTEXT
```

File: Makefile [hunk 3]
```diff
@@ -161,6 +161,8 @@ fmt: ## Check formatting of go sources.
 	@$(MAKE) go.init
 	@$(MAKE) go.fmt
 
+gen.codegen: codegen
+
```
Comment (BlaineEXE) [inline Makefile:165]: Is there a reason this one is separated by an empty line but others arent?

**Review by BlaineEXE — CHANGES_REQUESTED** [review]

**Comment by BlaineEXE** [comment]
Please also squash the commits. The last commit has a dependency on all the prior ones, so I don't see a procedural benefit to keeping them separate if we need to revert any single of the targets.

File: Makefile (hunk 3 above)
Comment (obnoxxx) [reply Makefile:165]: @BlaineEXE wrote:
> Is there a reason this one is separated by an empty line but others arent?


No particular reason.

Will make this systematic as well.

**Comment by obnoxxx** [comment]
@BlaineEXE  wrote:
> Please also squash the commits. The last commit has a dependency on all the prior ones, so I don't see a procedural benefit to keeping them separate if we need to revert any single of the targets.

Agreed. I squashed all commits into one, including your committed suggestions.

File: Makefile (hunk 3 above)
Comment (obnoxxx) [reply Makefile:165]: done.

**Review by subhamkrai — COMMENTED** [review]
Is CI already updated to this?

**Comment by subhamkrai** [comment]
> Is CI already updated to this?

also, do we have some place to see what is the right format? I'm not blocking this PR but just thinking about what advantages we are getting. Is this followed by most projects we are trying to follow?

**Review by BlaineEXE — COMMENTED** [review]
I think `helm-docs` also counts as a generator, and IMO `gen.docs` and `docs` should probably be colocated with that target. I'm not sure why they're separated in the current file

File: Makefile [hunk 4]
```diff
@@ -218,11 +226,16 @@ docs-preview: ## Preview the documentation through mkdocs
 docs-build:  ## Build the documentation to the `site/` directory
 	mkdocs build --strict
 
+gen.crd-docs: generate-docs-crds
+
 generate-docs-crds: ## Build the documentation for CRD
 	@GOBIN=$(GOBIN) build/crds/generate-crd-docs.sh
 
+generate: gen.codegen gen.crds gen.rbac gen.docs gen.crd-docs ##generate:  invoke all generators
```
[the remark this answers is not in the document]
Comment (obnoxxx) [reply Makefile]: done

**Comment by obnoxxx** [comment]
@subhamkrai  wrote:
> > Is CI already updated to this?

The workflow files do not use the new aliases (yet).

If that's what you are suggesting, I can change workflows to use the new aliases instead of the old targets.


> also, do we have some place to see what is the right format?

I am not claiming there is a right and a wrong naming.

The existing targets remain valid and just get som new, more  systematically named, aliases.





 I'm not blocking this PR but just thinking about what advantages we are getting. Is this followed by most projects we are trying to follow?

I am not sure that this is a standard pattern used by many projects.
I was just followin @BlaineEXE 's suggestion to establish a more systematic naming for all targets that are generators.

**Comment by subhamkrai** [comment]
> @subhamkrai wrote:
> 
> > > Is CI already updated to this?
> 
> The workflow files do not use the new aliases (yet).
> 
> If that's what you are suggesting, I can change workflows to use the new aliases instead of the old targets.

No, no current workflows works for me.
> 
> > also, do we have some place to see what is the right format?
> 
> I am not claiming there is a right and a wrong naming.
> 
> The existing targets remain valid and just get som new, more systematically named, aliases.
> 
> I'm not blocking this PR but just thinking about what advantages we are getting. Is this followed by most projects we are trying to follow?
> 
> I am not sure that this is a standard pattern used by many projects. I was just followin @BlaineEXE 's suggestion to establish a more systematic naming for all targets that are generators.

I don't have strong opinion here, if we feel this is the right way I'm good with the changes then. I'll approve

**Review by subhamkrai — APPROVED** [review]
Approving the PR as I'll be on PTO from tomorrow I don''t want to hold this. Thanks

File: Makefile [hunk 5]
```diff
@@ -218,11 +222,16 @@ docs-preview: ## Preview the documentation through mkdocs
 docs-build:  ## Build the documentation to the `site/` directory
 	mkdocs build --strict
 
+gen.crd-docs: generate-docs-crds
+
```
Comment (BlaineEXE) [inline Makefile:226]:
```suggestion
gen.crd-docs: generate-docs-crds
```

File: Makefile [hunk 6]
```diff
@@ -178,13 +179,16 @@ distclean: clean ## Remove all files that are created by building or configuring
 prune: ## Prune cached artifacts.
 	@$(MAKE) -C images prune
 
+gen.docs: docs
 docs: helm-docs
+gen.crds: crds
 
 crds: $(CONTROLLER_GEN) $(YQ)
```
Comment (BlaineEXE) [inline Makefile:186]:
```suggestion
gen.crds: crds
crds: $(CONTROLLER_GEN) $(YQ)
```

File: Makefile [hunk 7]
```diff
@@ -178,13 +179,16 @@ distclean: clean ## Remove all files that are created by building or configuring
 prune: ## Prune cached artifacts.
 	@$(MAKE) -C images prune
 
+gen.docs: docs
 docs: helm-docs
```
Comment (BlaineEXE) [inline Makefile:183]: Move to matching `helm-docs:` target.

**Review by BlaineEXE — CHANGES_REQUESTED** [review]

File: Makefile (hunk 7 above)
Comment (obnoxxx) [reply Makefile]: done

File: Makefile (hunk 5 above)
Comment (obnoxxx) [reply Makefile]: done

File: Makefile (hunk 6 above)
Comment (obnoxxx) [reply Makefile]: done

**Review by BlaineEXE — APPROVED** [review]

**Pull request merged**